### PR TITLE
Added Pub/Sub samples.

### PR DESCRIPTION
**Pull request by jmdobry**

### CODE REVIEW

File: pubsub/subscription.js [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+
+exports.createTopic = createTopic;
+exports.publish = publish;
+exports.getAllTopics = getAllTopics;
+exports.getAllSubscriptions = getAllSubscriptions;
+exports.pullMessages = pullMessages;
+exports.subscribe = subscribe;
+
+if (module === require.main) {
+  var _subscription
```
Comment (JustinBeckwith) [inline pubsub/subscription.js]: I'm surprised our jshint config doesn't require trailing semicolons

**Comment by JustinBeckwith** [comment]
LGTM

File: pubsub/subscription.js (hunk 1 above)
Comment (jmdobry) [inline pubsub/subscription.js]: It does. I'm not done.

**Comment by jmdobry** [comment]
Ready for review.

File: logging/export.js [hunk 2]
```diff
@@ -14,32 +14,38 @@
...
 
+// If you don't set the environment variables, then you can modify this file
+// to set the values
 projectId = projectId || '<your-project-id>';
 keyFilename = keyFilename || '/path/to/keyfile.json';
 
+// Provide projectId and authentication to gcloud
 var gcloud = require('gcloud')({
   projectId: projectId,
   keyFilename: keyFilename
```
Comment (jonparrott) [inline logging/export.js]: keyfilename is optional, as it can automatically use application default credentials.

File: pubsub/iam.js [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+'use strict';
+
+var async = require('async');
+var utils = require('./subscription');
```
Comment (jonparrott) [inline pubsub/iam.js]: this seems weird.

File: pubsub/iam.js [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+function getTopicPolicy(topic, callback) {
+  // Retrieve the IAM policy for the provided topic
+  topic.iam.getPolicy(callback);
+}
+// [END get_topic_policy]
+
+// [START get_subscription_policy]
+function getSubscriptionPolicy(subscription, callback) {
+  // Retrieve the IAM policy for the provided subscription
+  subscription.iam.getPolicy(callback);
```
Comment (jonparrott) [inline pubsub/iam.js]: Hmm.. this seems weird too. What are you actually showing here? I think the 'meaty' bit is what's passed into callback. Can you find a way to show both the call and the callback at once?

File: pubsub/iam.js [hunk 5]
```diff
@@ -0,0 +1,183 @@
...
+exports.testTopicPermissions = testTopicPermissions;
+exports.testSubscriptionPermissions = testSubscriptionPermissions;
+exports.runSample = runSample;
+
+function runSample(callback) {
+  var _subscription;
+  var _topic;
+  // Gather responses
+  var responses = [];
+  async.waterfall([
```
Comment (jonparrott) [inline pubsub/iam.js]: I personally find this part really hard to follow. I don't know if you plan on showing it in the docs.

File: pubsub/iam.js [hunk 6]
```diff
@@ -0,0 +1,183 @@
...
+      _topic = topic;
+      responses.push([topic, apiResponse]);
+      console.log('created topic');
+      console.log('get topic IAM policy...');
+      getTopicPolicy(topic, cb);
+    },
+    function (policy, apiResponse, cb) {
+      responses.push([policy, apiResponse]);
+      console.log('got topic policy', policy);
+    // IAM server is down?
```
Comment (jonparrott) [inline pubsub/iam.js]: Is this commented out code necessary?

File: pubsub/subscription.js [hunk 7]
```diff
@@ -0,0 +1,246 @@
...
+  // Grab paginated topics
+  pubsub.getTopics(function (err, topics, nextQuery) {
+    // Quit on error
+    if (err) {
+      return callback(err);
+    }
+    // There is another page of topics
+    if (nextQuery) {
+      // Grab the remaining pages of topics recursively
+      return getAllTopics(function (err, _topics) {
```
Comment (jonparrott) [inline pubsub/subscription.js]: clever. :)

File: pubsub/subscription.js [hunk 8]
```diff
@@ -0,0 +1,246 @@
...
+        // new messages are available. Returns if timeout is reached.
+        returnImmediately: false
+      };
+      // Pull any messages on the subscription
+      subscription.pull(options, function (err, messages) {
+        if (err) {
+          return callback(err);
+        }
+
+        // Do something with messages here?
```
Comment (jonparrott) [inline pubsub/subscription.js]: maybe show a per-message callback or event stream? Like we do in getting-started?

File: pubsub/iam.js (hunk 5 above)
Comment (jmdobry) [inline pubsub/iam.js]: not shown in the docs
